src/test/scala/org/ergoplatform/nodeView/history/VerifyADHistorySpecification.scala

```diff
@@ -11,31 +11,58 @@ import scala.util.{Random, Try}
 
 class VerifyADHistorySpecification extends HistorySpecification {
 
-  private def genHistory(height: Int = 0): ErgoHistory = {
+  private def genHistory(height: Int = 0, minFullHeight: Option[Int] = Some(0)): (ErgoHistory, Seq[ErgoFullBlock]) = {
     val inHistory = generateHistory(verifyTransactions = true, StateType.Digest, PoPoWBootstrap = false, BlocksToKeep)
-    if (height > 0) applyChain(inHistory, genChain(height, inHistory))
-    else inHistory
+    minFullHeight.foreach(h => inHistory.pruningProcessor.minimalFullBlockHeightVar = h)
+
+    if (height > 0) {
+      val chain = genChain(height, inHistory)
+      (applyChain(inHistory, chain), chain)
+    } else {
+      (inHistory, Seq.empty)
+    }
   }
 
-  property("missedModifiersForFullChain") {
-    var history = genHistory()
-    val chain = genChain(BlocksToKeep)
+  property("should not be able to apply blocks older than blocksToKeep") {
```

Comment:
Maybe we could introduce alias methods for the history like this:

```
property("should not be able to apply blocks older than blocksToKeep") {
    var history = genHistory()._1
    def appendHistory(modifier: ErgoPersistentModifier) =  history.append(modifier).get._1
    def bestFullBlockOpt = history.bestFullBlockOpt
    def bestFullBlockHeader = bestFullBlockOpt.get.header

    bestFullBlockOpt shouldBe None

    val chain = genChain(BlocksToKeep * 2)

    history = applyHeaderChain(history, HeaderChain(chain.map(_.header)))
    history.bestHeaderOpt shouldBe Some(chain.last.header)
    bestFullBlockOpt shouldBe None

    val fullBlocksToApply = chain.tail
    history.pruningProcessor.updateBestFullBlock(fullBlocksToApply(BlocksToKeep - 1).header)

    history.applicable(chain.head.blockTransactions) shouldBe false

    history = appendHistory(fullBlocksToApply.head.blockTransactions)
    bestFullBlockOpt shouldBe None
    history = appendHistory(fullBlocksToApply.head.aDProofs.get)
    bestFullBlockHeader shouldBe fullBlocksToApply.head.header

    history.applicable(chain.head.blockTransactions) shouldBe false

    fullBlocksToApply.tail.foreach { f =>
      history = appendHistory(f.blockTransactions)
      history = appendHistory(f.aDProofs.get)
    }
    bestFullBlockHeader shouldBe fullBlocksToApply.last.header

    //block transactions should be already pruned
    history.contains(fullBlocksToApply.head.blockTransactions) shouldBe false
    //block transactions should not be able to apply since they are too far back in history
    history.applicable(fullBlocksToApply.head.blockTransactions) shouldBe false
  }
```